l1-contracts/test/fees/MinimalFeeModel.sol

```diff
@@ -3,65 +3,120 @@
+// is simpler to compare etc.
 
 contract MinimalFeeModel is TimeFns {
   using FeeMath for OracleInput;
   using FeeMath for uint256;
+  using SlotLib for Slot;
 
   // This is to allow us to use the cheatcodes for blobbasefee as foundry does not play nice
   // with the block.blobbasefee value if using cheatcodes to alter it.
   Vm internal constant VM = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
 
+  uint256 internal constant BLOB_GAS_PER_BLOB = 2 ** 17;
+  uint256 internal constant GAS_PER_BLOB_POINT_EVALUATION = 50_000;
+
   // The L1 base fees are "fixed" for a period of 5 slots.
   // and is lagging 5 slots behind the current slot.
   Slot public constant LIFETIME = Slot.wrap(5);
   Slot public constant LAG = Slot.wrap(2);
   Timestamp public immutable GENESIS_TIMESTAMP;
 
   uint256 public populatedThrough = 0;
-  mapping(uint256 _slotNumber => DataPoint _dataPoint) public dataPoints;
+  mapping(uint256 slotNumber => FeeHeader feeHeader) public feeHeaders;
 
-  L1BaseFees public l1BaseFees;
+  L1GasOracleValues public l1BaseFees;
 
   constructor(uint256 _slotDuration, uint256 _epochDuration) TimeFns(_slotDuration, _epochDuration) {
     GENESIS_TIMESTAMP = Timestamp.wrap(block.timestamp);
-    dataPoints[0] = DataPoint({provingCostNumerator: 0, feeAssetPriceNumerator: 0});
+    feeHeaders[0] = FeeHeader({
+      excess_mana: 0,
+      fee_asset_price_numerator: 0,
+      mana_used: 0,
+      proving_cost_per_mana_numerator: 0
+    });
 
-    l1BaseFees.pre = BaseFees({baseFee: 1 gwei, blobFee: 1});
-    l1BaseFees.post = BaseFees({baseFee: block.basefee, blobFee: _getBlobBaseFee()});
-    l1BaseFees.slotOfChange = LIFETIME;
+    l1BaseFees.pre = L1Fees({base_fee: 1 gwei, blob_fee: 1});
+    l1BaseFees.post = L1Fees({base_fee: block.basefee, blob_fee: _getBlobBaseFee()});
+    l1BaseFees.slot_of_change = LIFETIME.unwrap();
+  }
+
+  function getL1GasOracleValues() public view returns (L1GasOracleValues memory) {
+    return l1BaseFees;
+  }
+
+  // For all of the estimations we have been using `3` blobs.
+  function manaBaseFeeComponents(uint256 _blobsUsed, bool _inFeeAsset)
+    public
+    view
+    returns (ManaBaseFeeComponents memory)
+  {
+    L1Fees memory fees = getCurrentL1Fees();
+    uint256 dataCost = Math.mulDiv(
+      _blobsUsed * BLOB_GAS_PER_BLOB, fees.blob_fee, FeeMath.MANA_TARGET, Math.Rounding.Ceil
+    );
+    uint256 gasUsed = FeeMath.L1_GAS_PER_BLOCK_PROPOSED + _blobsUsed * GAS_PER_BLOB_POINT_EVALUATION
+      + FeeMath.L1_GAS_PER_EPOCH_VERIFIED / EPOCH_DURATION;
+    uint256 gasCost = Math.mulDiv(gasUsed, fees.base_fee, FeeMath.MANA_TARGET, Math.Rounding.Ceil);
+    uint256 provingCost = getProvingCost();
+
+    uint256 congestionMultiplier = FeeMath.congestionMultiplier(calcExcessMana());
+
+    uint256 total = dataCost + gasCost + provingCost;
+    uint256 congestionCost =
+      (total * congestionMultiplier / FeeMath.MINIMUM_CONGESTION_MULTIPLIER) - total;
```

Comment:
Just curious, why don't we need muldiv here?

Reply:
This is a muldiv just not rounding up. We are rounding up on the others as they cover the "real" cost components and we want to make sure that they are covered. Should probably do it using the same muldiv function though, just for clarity